### PR TITLE
Colab demo QOL improvements

### PR DESCRIPTION
**Pull request by neubig**
# Description

This makes two improvements to the Colab demo:

1. It makes the demo runnable by default on T4 GPUs on colab, so you don't need to pay money to run by default
2. It reduces the size of the validation set, so that validation doesn't take quite as long with smaller batch sizes

In order to do 2, I introduced functionality to vary the maximum size of the train, dev, and test sets.

# Blocked by

- NA

### CODE REVIEW

**Review by viswavi — APPROVED** [review]
LGTM
